src/inchat-octave.js

```diff
@@ -1,30 +1,235 @@
-let fs = require('fs');
+const fs = require('fs');
+const { exec } = require('child_process');
+const util = require('util');
+const download = require('./download');
 
-// Function to clear out workspace `.mat` files 
+const lengthMaxBotMessages = 1000; // max message length
+
+// Define the path variables for inchat octave
+const ic_octave_folder = './inchat_octave';
+const ic_octave_workspaces = util.format('%s/workspaces', ic_octave_folder);
+const ic_octave_out_file = util.format('%s/bot_out.txt', ic_octave_folder);
+const ic_octave_user_code = util.format('%s/user_code.m', ic_octave_folder);
+const ic_octave_printout = util.format('%s/user_printout.png', ic_octave_folder);
+const ic_octave_timeout = 5000; // time in ms
+
+// Load in illegal use functions for inchat octave and compile them as a regexp
+const illegal_read = fs.readFileSync(util.format('%s/illegal_phrases', ic_octave_folder), 'utf8');
+const illegal_use_regexp = new RegExp(`(${illegal_read.replace(/\n/g, ')|(')})`);
+
+function octaveRun(msg, operation) {
+    // Figure out the workspace filename for this user
+    const user_id = `${msg.author.username}#${msg.author.discriminator}`;
+    const user_work_file = `${ic_octave_workspaces}/${user_id}.mat`;
+
+    const oct_call_regexp = /!o\w+\s*(?:```matlab)?([\s\S.]*[^`])(?:```)?$/;
+    const oct_call_tokens = msg.content.match(oct_call_regexp);
+
+    // Grab the users commands
+    const code = oct_call_tokens[1];
+
+    // Check for illegal command usage and warn against it...
+    const found_illegal_match = code.match(illegal_use_regexp);
+    if (found_illegal_match) {
+        msg.channel.send(`Someone was being naughty <@${process.env.OWNER_ID}>`);
+        return;
+    }
+
+    // Write the users code command to a file. continue execution if it works
+    fs.writeFile(ic_octave_user_code, code, (err) => {
```

Comment:
Shouldn't these files be cleaned up after execution 👀

Reply:
> yes! Thanks for pointing out